webportal/src/app/user/user-register/user-register.component.js

```diff
@@ -67,4 +90,130 @@ $(document).ready(() => {
       });
     });
   });
+
+  $('#form-update-account').on('submit', (e) => {
+    e.preventDefault();
+    const username = $('#form-update-account :input[name=username]').val();
+    const password = $('#form-update-account :input[name=password]').val();
+    const admin = $('#form-update-account :input[name=admin]').is(':checked') ? true : false;
+    userAuth.checkToken((token) => {
+      $.ajax({
+        url: `${webportalConfig.restServerUri}/api/v1/user`,
+        data: {
+          username,
+          password,
+          admin: admin,
+          modify: true,
+        },
+        type: 'PUT',
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+        dataType: 'json',
+        success: (data) => {
+          if (data.error) {
+            alert(data.message);
+          } else {
+            if (admin) {
+              $.ajax({
+                url: `${webportalConfig.restServerUri}/api/v1/user/${username}/virtualClusters`,
+                data: {
+                  virtualClusters: '',
+                },
+                type: 'PATCH',
+                headers: {
+                  Authorization: `Bearer ${token}`,
+                },
+                dataType: 'json',
+                success: (updateVcData) => {
+                  $('#form-update-account').trigger('reset');
+                  if (updateVcData.error) {
+                    alert(updateVcData.message);
+                  } else {
+                    alert('Update user basic information successfully');
+                  }
+                },
+                error: (xhr, textStatus, error) => {
+                  $('#form-update-account').trigger('reset');
+                  const res = JSON.parse(xhr.responseText);
+                  alert(res.message);
+                },
+              });
+            } else {
+              $('#form-update-account').trigger('reset');
+              alert('Update user basic information successfully');
+            }
+          }
+        },
+        error: (xhr, textStatus, error) => {
+          $('#form-update-account').trigger('reset');
+          const res = JSON.parse(xhr.responseText);
+          alert(res.message);
+        },
+      });
+    });
+  });
+
+  $('#form-update-virtual-cluster').on('submit', (e) => {
+    e.preventDefault();
+    const username = $('#form-update-virtual-cluster :input[name=username]').val();
+    const virtualCluster = $('#form-update-virtual-cluster :input[name=virtualCluster]').val();
+    userAuth.checkToken((token) => {
+      $.ajax({
+        url: `${webportalConfig.restServerUri}/api/v1/user/${username}/virtualClusters`,
+        data: {
+          virtualClusters: virtualCluster,
+        },
+        type: 'PATCH',
```

Comment:
`PATCH` may not be supported in jquery or edge, e.g. [ref1](https://stackoverflow.com/questions/13642044/does-the-jquery-ajax-call-support-patch), [ref2](https://stackoverflow.com/questions/11461414/ajax-json-doesnt-get-sent-in-patch-only).

Reply:
Thanks! I can use PUT instead.